cockatrice/src/remoteclient.cpp

```diff
@@ -301,21 +319,51 @@ void RemoteClient::readData()
     } while (!inputBuffer.isEmpty());
 }
 
+void RemoteClient::websocketMessageReceived(const QByteArray &message)
+{
+    lastDataReceived = timeRunning;
+    ServerMessage newServerMessage;
+    newServerMessage.ParseFromArray(message.data(), message.length());
+#ifdef QT_DEBUG
+    qDebug() << "IN" << messageLength << QString::fromStdString(newServerMessage.ShortDebugString());
+#endif
+    processProtocolItem(newServerMessage);
+}
+
 void RemoteClient::sendCommandContainer(const CommandContainer &cont)
 {
-    QByteArray buf;
-    unsigned int size = cont.ByteSize();
+
+    auto size = static_cast<unsigned int>(cont.ByteSize());
 #ifdef QT_DEBUG
     qDebug() << "OUT" << size << QString::fromStdString(cont.ShortDebugString());
 #endif
-    buf.resize(size + 4);
-    cont.SerializeToArray(buf.data() + 4, size);
-    buf.data()[3] = (unsigned char)size;
-    buf.data()[2] = (unsigned char)(size >> 8);
-    buf.data()[1] = (unsigned char)(size >> 16);
-    buf.data()[0] = (unsigned char)(size >> 24);
 
-    socket->write(buf);
+    QByteArray buf;
+    if (usingWebSocket) {
+        buf.resize(size);
+        cont.SerializeToArray(buf.data(), size);
+        websocket->sendBinaryMessage(buf);
+    } else {
+        buf.resize(size + 4);
+        cont.SerializeToArray(buf.data() + 4, size);
+        buf.data()[3] = (unsigned char)size;
+        buf.data()[2] = (unsigned char)(size >> 8);
+        buf.data()[1] = (unsigned char)(size >> 16);
+        buf.data()[0] = (unsigned char)(size >> 24);
+
+        socket->write(buf);
+    }
+}
+
+void RemoteClient::connectToHost(const QString &hostname, unsigned int port)
+{
+    usingWebSocket = port == 443 || port == 80 || port == 4748 || port == 8080;
```

Comment:
maybe it's easier to think that anything but 4747 is using websockets?
`usingWebSocket = port != 4747;`

Reply:
Was thinking the same thing.